model/user.go

```diff
@@ -745,6 +751,23 @@ func (u *User) IsRemote() bool {
 	return u.RemoteId != nil && *u.RemoteId != ""
 }
 
+// GetProp fetches a prop value by name.
+func (u *User) GetProp(name string) (string, bool) {
+	val, ok := u.Props[name]
+	return val, ok
+}
+
+// SetProp sets a prop value by name, creating the map if nil.
+// Not thread safe.
+func (u *User) SetProp(name string, value string) string {
```

Comment:
Do we actually need to return the previous value? If so we should probably document it.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/17347/commits/0b1337e31f9e880ac1098045a7f34c9adf44ce07